src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py

```diff
@@ -150,18 +152,25 @@ def id(self):
         # Assume md5 is good enough for now
         myhash = md5()
         spacer = ';'  # character not present in any field
-        myhash.update(self['RequestName'] + spacer)
```

Comment:
A possible suggestion I would have would be to have a clear separation of how this `id` is calculated in python2 and in python3, something like:
```
if PY2:
    same code as before
else:
    new code
```

I believe it would be more clear, however it would have more lines of code and some (almost) code duplication.
So, in the end I think this approach looks better (but if you see more value on the approach I mentioned, go for it!).

Reply:
Have a look at the new proposal, this should be a bit better, I hope

Reply:
That's a very original idea! However, I'm not very comfortable with this function change and I think it can add some confusion as well, especially because those 2 methods have become very popular and largely used in our system (I believe not yet by external services).

Actually, why don't we get a third opinion here? I like it very much, but I also wanted to keep it as simple as possible, which might not be the case anymore (or it's all in my head ;)). @khurtado @todor-ivanov could one of you please share your opinion with us?

Reply:
I like the `encodeUnicodeToBytes` method! It looks very similar to the HTCondor `classad.quote` and `classad.unquote` methods to handle "condor strings" to python strings/bytes and the other way around. I would probably prefer to hide the `condition` parameter in the method and import `Utils.PythonVersion`  in `Utils.Utilities` and use that information directly for the the encodeUnicodeToBytes method, if possible.

Reply:
At the beginning `encodeUnicodeToBytes` / `decodeBytesToUnicode` are conceived to give the same behavior in both py2 and py3 and we are already relying on this consistency, so i would not hide any check on PY2/PY3 inside them (examples: there are multiple occasions where we are using these function outside of `if PY3:`).

Here we want something different, we do not want consistency, we want to perform the encoding only with PY3 without being too-verbose. I understand that changing a function that is widespread may not be nice, what about a new couple of functions `encodeUnicodeToBytesConditional` / `decodeBytesToUnicodeConditional`, as just pushed?

Better name or different proposals are appreciated!

Reply:
This looks better in my opinion. Can you please just add a docstring to those new function saying something like: "condition has to be an expression evaluating to True/False, usually PY2 or PY3 though"?

Reply:
Sure! squashed and pushed.